src/commands/flinkStatements.ts

```diff
@@ -24,8 +43,143 @@ export async function viewStatementSqlCommand(statement: FlinkStatement): Promis
+  const computePool = await flinkComputePoolQuickPick();
+  if (!computePool) {
+    logger.error("sumbitFlinkStatementCommand", "computePool is undefined");
+    return;
+  }
+
+  // 4. Choose the current / default database / aka kafka cluster
+  // within the environment.
+  const currentDatabaseKafkaCluster: KafkaCluster | undefined = await kafkaClusterQuickPick({
+    placeHolder: "Select the Kafka cluster to use as the default database for the statement",
+    filter: (cluster: KafkaCluster) => {
+      if (cluster.environmentId !== computePool.environmentId) {
+        return false;
+      }
+      // Any survivors matching environmentId check should be CCloudKafkaClusters,
+      // since the compute pool is ccloud.
+      const ccloudCluster = cluster as CCloudKafkaCluster;
+      return (
+        ccloudCluster.provider === computePool.provider &&
+        ccloudCluster.region === computePool.region
+      );
+    },
+  });
+  if (!currentDatabaseKafkaCluster) {
+    logger.error("sumbitFlinkStatementCommand", "currentDatabaseKafkaCluster is undefined");
+    return;
+  }
+  const currentDatabase = currentDatabaseKafkaCluster.name;
+
+  // 5. Prep to submit, submit.
+  const submission: IFlinkStatementSubmitParameters = {
+    statement,
+    statementName,
+    computePool,
+    properties: new FlinkSpecProperties({
+      currentDatabase,
+      currentCatalog: computePool.environmentId,
+      // TODO get at the user's local timezone
+      // localTimezone: "GMT-04:00",
+    }),
+  };
+
+  try {
+    const restResponse = await submitFlinkStatement(submission);
+    const newStatement = restFlinkStatementToModel(restResponse);
+
+    if (newStatement.status.phase === FAILED_PHASE) {
+      // Immediate failure of the statement. User gave us something
+      // bad, like perhaps a bad table / column name, etc..
+
+      logUsage(UserEvent.FlinkStatementAction, {
+        action: "submit_failure",
+        compute_pool_id: computePool.id,
+        failure_reason: newStatement.status.detail,
+      });
+
+      await showErrorNotificationWithButtons(
+        `Error submitting statement: ${newStatement.status.detail}`,
+      );
+      return;
+    }
+
+    logUsage(UserEvent.FlinkStatementAction, {
+      action: "submit_success",
+      sql_kind: newStatement.sqlKind,
+      compute_pool_id: computePool.id,
+    });
+
+    // Refresh the statements view onto the compute pool in question,
+    // which will then show the new statement.
+    await selectPoolForStatementsViewCommand(computePool);
+
+    // TODO indicate to the view to highlight the new statement
+    // (similar to how we did we creating new schema subjects or versions)
+    // Something like:
+    // await FlinkStatementsViewProvider.getInstance().revealStatement(newStatement);
+
+    // TODO open up statement results view (invoke Rohit work here once both in main)
+  } catch (err) {
+    logError(err, "Submit Flink statement unexpected error");
```

Comment:
Do we want this to go to Sentry? Will need some kind of context passed if so, like:
```suggestion
    logError(err, "Submit Flink statement unexpected error", { extra: { computePoolId: ___ } });
```